### PR TITLE
KAZOO-4237 move outbound faxes to modb after process

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/fax/src/fax_maintenance.erl [hunk 1]
```diff
@@ -338,3 +341,77 @@ update_job(JobID, State, JObj) ->
...
+            case migrate_outbound_faxes(Docs, []) of
+                [] -> io:format("finished crawling faxes db~n");
+                Result -> migrate_outbound_faxes([{'startkey', next_key(lists:last(Result))}
+                                                  ,{'limit', props:get_value('limit', Options)}
+                                                 ])
+            end
+    end.
+
+-spec migrate_outbound_faxes(wh_json:objects(), list()) -> list().
+migrate_outbound_faxes([], Acc) -> lists:reverse(Acc);
```
Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: Why reverse it if you just want the last item just a couple lines above?

File: applications/fax/src/fax_maintenance.erl [hunk 2]
```diff
@@ -338,3 +341,77 @@ update_job(JobID, State, JObj) ->
...
+    FromDB = wh_doc:account_db(JObj),
+    AccountId = wh_doc:account_id(JObj),
+    AccountMODb = kazoo_modb:get_modb(AccountId, Year, Month),
+    kazoo_modb:create(AccountMODb),
+    ToDB = wh_util:format_account_modb(AccountMODb, 'encoded'),
+    ToId = <<(wh_util:to_binary(Year))/binary
+              ,(wh_util:pad_month(Month))/binary
+              ,"-"
+              ,FromId/binary
+            >>,
```
Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: You can create these binaries using the `?MATCH_MODB_…` suite. See kazoo_modb_util.erl for example.

File: applications/fax/src/fax_worker.erl [hunk 3]
```diff
@@ -686,16 +684,32 @@ set_default_update_fields(JObj) ->
...
+    FromDB = wh_doc:account_db(JObj),
+    AccountId = wh_doc:account_id(JObj),
+    AccountMODb = kazoo_modb:get_modb(AccountId, Year, Month),
+    kazoo_modb:create(AccountMODb),
+    ToDB = wh_util:format_account_modb(AccountMODb, 'encoded'),
+    ToId = <<(wh_util:to_binary(Year))/binary
+              ,(wh_util:pad_month(Month))/binary
+              ,"-"
+              ,FromId/binary
+            >>,
```
Comment (fenollp) [inline applications/fax/src/fax_worker.erl]: same here

File: core/whistle-1.0.0/src/wh_util.erl [hunk 4]
```diff
@@ -911,6 +913,18 @@ to_boolean(<<"false">>) -> 'false';
 to_boolean("false") -> 'false';
 to_boolean('false') -> 'false'.
 
+-spec to_date(binary() | string() | integer()) -> wh_date().
+to_date(X) when is_integer(X) ->
+    {Date, _ } = calendar:gregorian_seconds_to_datetime(X),
+    Date;
+to_date(X) when is_binary(X) -> to_date(to_integer(X));
+to_date(X) when is_list(X) -> to_date(to_integer(X)).
```
Comment (fenollp) [inline core/whistle-1.0.0/src/wh_util.erl]: Why not just reuse `to_datetime/1`?

File: applications/fax/src/fax_maintenance.erl (hunk 1 above)
Comment (lazedo) [inline applications/fax/src/fax_maintenance.erl]: because of this  [DocId | Acc] on line 375

File: core/whistle-1.0.0/src/wh_util.erl (hunk 4 above)
Comment (lazedo) [inline core/whistle-1.0.0/src/wh_util.erl]: makes sense

File: applications/fax/src/fax_maintenance.erl [hunk 5]
```diff
@@ -338,3 +341,71 @@ update_job(JobID, State, JObj) ->
...
+                                    ,{'limit', props:get_value('limit', Options)}
+                                   ])
+    end.
+
+-spec migrate_outbound_faxes(wh_json:objects(), api_binary()) -> api_binary().
+migrate_outbound_faxes([], Acc) -> Acc;
+migrate_outbound_faxes([JObj | JObjs], Acc) ->
+    DocId = wh_doc:id(JObj),
+    maybe_migrate_outbound_fax(DocId),
+    migrate_outbound_faxes(JObjs, DocId).
```
Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: Where did `[DocId | Acc]` go?

Comment (lazedo) [inline applications/fax/src/fax_maintenance.erl]: i changed because there is no need to accumulate, we just need the last.

Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: Line 370: s/Acc/_Acc/

File: applications/fax/src/fax_maintenance.erl [hunk 6]
```diff
@@ -338,3 +341,71 @@ update_job(JobID, State, JObj) ->
...
+    AccountMODb = kazoo_modb:get_modb(AccountId, Year, Month),
+    kazoo_modb:create(AccountMODb),
+    ToDB = wh_util:format_account_modb(AccountMODb, 'encoded'),
+    ToId = ?MATCH_MODB_PREFIX(wh_util:to_binary(Year), wh_util:pad_month(Month),FromId),
+    case couch_mgr:move_doc(FromDB, FromId, ToDB, ToId, ['override_existing_document']) of
+        {'ok', _} -> io:format("document ~s/~s moved to ~s/~s~n", [FromDB, FromId, ToDB, ToId]);
+        {'error', _E} -> io:format("error ~p moving document ~s/~s to ~s/~s~n", [_E, FromDB, FromId, ToDB, ToId])
+    end.
+
+-spec next_key(binary()) -> ne_binary().
```
Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: Needs to handle case `undefined`.

Comment (lazedo) [inline applications/fax/src/fax_maintenance.erl]: no it doesn't.

File: applications/fax/src/fax_maintenance.erl [hunk 7]
```diff
@@ -338,3 +341,71 @@ update_job(JobID, State, JObj) ->
...
+    migrate_outbound_faxes([{'limit', Number}]);
+migrate_outbound_faxes(Options) ->
+    case couch_mgr:all_docs(?WH_FAXES_DB, Options) of
+        {'error', _E} ->
+            io:format("failed to crawl faxes db: ~p~n", [ _E]);
+        {'ok', []} ->
+            io:format("finished crawling faxes db~n");
+        {'ok', Docs} ->
+            Last = migrate_outbound_faxes(Docs, 'undefined'),
+            migrate_outbound_faxes([{'startkey', next_key(Last)}
```
Comment (fenollp) [inline applications/fax/src/fax_maintenance.erl]: `Last` can be `undefined` here, no?

Comment (lazedo) [inline applications/fax/src/fax_maintenance.erl]: no it can't

**Pull request merged**